A test test_tg01fd.test4_ag08bd failure

Hi Slycot Team, I am sharing the log of test test_tg01fd.test4_ag08bd  over Arch Linux system as [suggests the maintainer](https://aur.archlinux.org/packages/python-slycot/#comment-831367). The [full log](https://gist.githubusercontent.com/carlosal1015/01caa7b3d6939318ca2235fc8674b6ad/raw/6bd085ff0f183867e0736b90d87c809be5c51fc2/python-slycot-0.4.0.0-3-log.txt) is here.

We are using:
- [`python-3.9.7-1-x86_64`](https://archlinux.org/packages/core/x86_64/python/)
- [`python-numpy-1.21.2-1-x86_64`](https://archlinux.org/packages/extra/x86_64/python-numpy)
- [`lapack-3.10.0-1-x86_64`](https://archlinux.org/packages/extra/x86_64/lapack)
- [`blas-3.10.0-1-x86_64`](https://archlinux.org/packages/extra/x86_64/blas)

```py
============================= test session starts ==============================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
rootdir: /tmp/makepkg/python-slycot/src
collected 108 items

slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_ab01.py . [  0%]
                                                                         [  0%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_ab08n.py . [  1%]
.                                                                        [  2%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_ag08bd.py . [  3%]
..F                                                                      [  6%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_analysis.py . [  7%]
..................                                                       [ 24%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_examples.py . [ 25%]
............                                                             [ 36%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_exceptions.py . [ 37%]
..                                                                       [ 38%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_mb.py . [ 39%]
.........                                                                [ 48%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_mc.py . [ 49%]
..                                                                       [ 50%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_sb.py . [ 51%]
............................                                             [ 77%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_sg02ad.py . [ 78%]
                                                                         [ 78%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_sg03ad.py . [ 79%]
..                                                                       [ 81%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_tb05ad.py . [ 82%]
.....                                                                    [ 87%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_td04ad.py . [ 87%]
......                                                                   [ 93%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_tg01ad.py . [ 94%]
                                                                         [ 94%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_tg01fd.py . [ 95%]
.                                                                        [ 96%]
slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_transform.py . [ 97%]
...                                                                      [100%]

=================================== FAILURES ===================================
___________________________ test_tg01fd.test4_ag08bd ___________________________

self = <slycot.tests.test_ag08bd.test_tg01fd testMethod=test4_ag08bd>

    def test4_ag08bd(self):
        """test [A-lambda*E,B;C,D]"""
    
        Af,Ef,nrank,niz,infz,kronr,infe,kronl = analysis.ag08bd(l=test1_l,n=test1_n,m=test1_m,p=test1_p,A=test1_A,E=test1_E,B=test1_B,C=test1_C,D=test1_D,equil=test1_equil, tol=test1_tol)
    
>       assert_almost_equal(Af, [[0.77045021]])
E       AssertionError: 
E       Arrays are not almost equal to 7 decimals
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 1.54090042
E       Max relative difference: 1.99999999
E        x: array([[-0.7704502]])
E        y: array([[0.7704502]])

slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_ag08bd.py:111: AssertionError
=========================== short test summary info ============================
FAILED slycot-0.4.0.0/_skbuild/linux-x86_64-3.9/setuptools/lib/slycot/tests/test_ag08bd.py::test_tg01fd::test4_ag08bd
======================== 1 failed, 107 passed in 1.32s =========================
```

--- a/slycot/tests/test_ag08bd.py
+++ b/slycot/tests/test_ag08bd.py
@@ -47,8 +47,8 @@ test1_D = np.array([[ 1,  2, -2],
                     [ 0,  0,  0]])
 
 
-class test_tg01fd(unittest.TestCase):
-    """ Verify ag08bd with input parameters according to example in documentation """
+class test_ag08bd(unittest.TestCase):
+    """Verify ag08bd with input parameters according to example in documentation."""
 
     def test1_ag08bd(self):
         """test [A-lambda*E]
@@ -108,8 +108,9 @@ class test_tg01fd(unittest.TestCase):
 
         Af,Ef,nrank,niz,infz,kronr,infe,kronl = analysis.ag08bd(l=test1_l,n=test1_n,m=test1_m,p=test1_p,A=test1_A,E=test1_E,B=test1_B,C=test1_C,D=test1_D,equil=test1_equil, tol=test1_tol)
 
-        assert_almost_equal(Af, [[0.77045021]])
-        assert_almost_equal(Ef, [[0.77045021]])
+        # Af-lambda*Ef==0. => lambda==1. => Finite Smith zero of S(lambda) == 1.
+        assert Af.shape == (1, 1)
+        assert_almost_equal(Af, Ef)
         assert_equal(nrank, 11)
         assert_equal(niz, 2)
         assert_equal(infz, [0,1])

